util/Logger.h

```diff
@@ -1,56 +1,292 @@
+            a problem.
+            For example a missing id that can be ignored, an extra item in a container.
+          - used to indicate non critcal degraded system state to the player.  Warnings may be
+            unavoidable due to system issues outside of the game's control.
+            For example warning that the sound system is unavailable.
+    info  - used to report normal game state and progress.  This should be the default level of
+            logging.  The level of detail and the volume should be low.  The number of log entries
+            should be low enough to not require truncating the log even for a complete game.  The
+            information should be of interest to all devs and interested players and of a nature
+            that people unfamiliar with the code can follow.
+            For example the unnamed logger reporting that turn X has started.
+            Another example the network logger reporting that the network connected/disconnected.
+    debug - used for low-level implementation or calculation details.  For a named logger this
+            level will probably only be turned on by devs working on that section of code.
+            This will be detailed and perhaps voluminous.  Debug messages may require familiarity
+            with the code to understand.
+            For example reporting that the network disconnected due to a client initiated shutdown
+            with a linger time of 30 ms before closing.
+    trace - used for the most detailed logging.  Trace should probably only be used with a named
+            logger, since the extreme detail will only be of interest to a developer working on a
+            specific section of code and without the additional filtering of the name the log
+            output will be flooded.  Setting all logger's thresholds to trace may generate
+            unreasonably large log files.
+            For example the network logger might log every message received/sent as it is
+            added to or removed from its internal queues.
+
+    Usage:
+
+    Generating Logs:
+
+    The default or unnamed logger works as an output stream, like std::cout as follows:
+
+    ErrorLogger() << "Put any streamable output here.  It will only be
+                  << "computed at the error threshold or greater.";
+    WarnLogger()  << "Put any streamable output here.  It will only be
+                  << "computed at the warn threshold or greater.";
+    InfoLogger()  << "Put any streamable output here.  It will only be
+                  << "computed at the info threshold or greater.";
+    DebugLogger() << "Put any streamable output here.  It will only be
+                  << "computed at the debug threshold or greater.";
+    TraceLogger() << "Put any streamable output here.  It will only be
+                  << "computed at the trace threshold or greater.";
+
+    The named loggers are created with:
+
+    CreateThreadedLogger(name_of_logger);
```

Comment:
Is it necessary to have this function be called before using a named logger? It's not clear where to put it, or if it's a problem to call it twice... or why it's "Threaded". Can't I just use DebugLogger("effects") and have it create that logger if it didn't already exist the first time something is logged to that name?